src/OrleansRuntime/Counters/SiloStatisticsManager.cs

```diff
@@ -12,18 +12,23 @@ internal class SiloStatisticsManager
         internal SiloPerformanceMetrics MetricsTable;
         private readonly Logger logger = LogManager.GetLogger("SiloStatisticsManager");
 
-        internal SiloStatisticsManager(GlobalConfiguration globalConfig, NodeConfiguration nodeConfig)
+        public SiloStatisticsManager(SiloInitializationParameters initializationParams)
         {
             MessagingStatisticsGroup.Init(true);
             MessagingProcessingStatisticsGroup.Init();
             NetworkingStatisticsGroup.Init(true);
-            ApplicationRequestsStatisticsGroup.Init(globalConfig.ResponseTimeout);
+            ApplicationRequestsStatisticsGroup.Init(initializationParams.GlobalConfig.ResponseTimeout);
             SchedulerStatisticsGroup.Init();
             StorageStatisticsGroup.Init();
             runtimeStats = new RuntimeStatisticsGroup();
-            logStatistics = new LogStatistics(nodeConfig.StatisticsLogWriteInterval, true);
-            MetricsTable = new SiloPerformanceMetrics(runtimeStats, nodeConfig);
-            countersPublisher = new CountersStatistics(nodeConfig.StatisticsPerfCountersWriteInterval);
+            this.logStatistics = new LogStatistics(initializationParams.NodeConfig.StatisticsLogWriteInterval, true);
```

Comment:
You're right. This PR already ended up being larger in scope than I had hoped, though. It's definitely not supposed to reach the logical conclusion of the direction it starts us in - just lays some groundwork and gives us something which can be reviewed a little more quickly than a big(er) bang PR.